text/0750-feature-flag-advisor.md

```diff
@@ -0,0 +1,183 @@
+    **Feature** **** **Flag** **** **Recommended** **** **User**
+    @aws-cdk/...                              true                         true
+  * @aws-cdk/...                              true                         false
+  * @aws-cdk/...                              true                         false
+
+Would you like to switch on all recommended flags? (Y/n)
+> Y
+Resynthesizing... 
+Here is the difference:
+- S3 Bucket:
+-    Properties:
+-      `BucketEncryption``:`` ``None`
++ S3 Bucket:
++    Properties:
++      BucketEncryption: 
++        `ServerSideEncryptionConfiguration``:`
++          ...
+```
+
+To switch the state of a specific feature flag, run `cdk flags --toggle="#FLAGNAME#"`. After showing the difference in CloudFormation templates, you will be asked if you would like to keep the changes or reject them.
+
+```
+> cdk flags --toggle="#FLAGNAME#"
+    Feature Flag                              Recommended                  User
+    @aws-cdk/...                              true                         true
+  * @aws-cdk/...                              true                         false
+  * @aws-cdk/...                              true                         false
+
+Would you like to switch the state of "#FLAGNAME#"? (Y/n)
+> Y
+Resynthesizing... 
+Here is the difference:
+- S3 Bucket:
+-    Properties:
+-      `BucketEncryption``:`` ``None`
++ S3 Bucket:
++    Properties:
++      BucketEncryption: 
++        `ServerSideEncryptionConfiguration``:`
++          ...
+
+Would you like to accept these changes? (Accept/reject)
+> reject
+- S3 Bucket:                                   --    
+-    Properties:                                |
+-      BucketEncryption:                        |    CURRENT VERSION 
+-        ServerSideEncryptionConfiguration:     |
+-          ...                                 --
++ S3 Bucket:                             --
++    Properties:                          |    INCOMING VERSION
++      BucketEncryption: None            --
+Resynthesizing... 
+Done.
+```
+
+## What are we trying to solve?
+
+Feature flags are an important tool for adding new bug fixes without affecting the current state of the application. However, feature flags often go unnoticed by the user. In order to change their configurations, users are required to manually update their `cdk.json` file, resulting in a time-consuming task. They can impact a user in many ways. For example, the @aws-cdk/aws-cloudfront:defaultSecurityPolicyTLSv1.2_2021 flag changed the default behavior to use the updated version TLSv1.2_2021, which is a more secure and support TLS version for CloudFront. Users who stuck with the previous version may have been exposed to security vulnerabilities and failed compliance tests. To remain Backwards Compatible, newly released feature flags are automatically disabled for existing projects, meaning that those users are required to constantly keep an eye on CDK releases to manually update their feature flags. 
```

Comment:
Honestly, a fraction of this can go into the introductory paragraph of the `flags` command as well. This is useful context! 

Something like (but not necessarily equal to)

> CDK uses feature flags to roll out changes that may affect existing infrastructure definitions. The `cdk flags` command will show you the available flags, their recommended values, and your currently configured values, along with commands to automatically toggle flags to their recommended states.